### PR TITLE
docs: update references to example repos that will move external

### LINKED ISSUE #2015: Update references to `mst-example-bookshop`
We should find anywhere we control references to the current `mst-example-bookshop`, and update it with wherever we land in #2014.

For places that we don't control, maybe we consider reaching out to folks, or putting up a note somewhere to flag the move.

### LINKED ISSUE #2019: Update references to `mst-example-boxes`
We should find anywhere we control references to the current `mst-example-bookshop`, and update it with wherever we land in #2018.

For places that we don't control, maybe we consider reaching out to folks, or putting up a note somewhere to flag the move.

### LINKED ISSUE #2022: Update references to `mst-example-redux-todomvc`
We should find anywhere we control references to the current `mst-example-redux-todomvc`, and update it with wherever we land in #2021 

For places that we don't control, maybe we consider reaching out to folks, or putting up a note somewhere to flag the move.

### LINKED ISSUE #2025: Update references to `mst-example-todomvc `
We should find anywhere we control references to the current `mst-example-todomvc `, and update it with wherever we land in #2024 

For places that we don't control, maybe we consider reaching out to folks, or putting up a note somewhere to flag the move.

### PR DESCRIPTION
**Pull request by coolsoftwaretyler**
This PR updates references to the four example repositories which I have moved to external repositories (@jamonholmgren also has access to these).

Closes #2015 

Closes #2019 

Closes #2022 

Closes #2025

Once we do this, we can remove those directories and all their monorepo configuration.

### CODE REVIEW

File: docs/concepts/async-actions.md [hunk 1]
```diff
@@ -21,8 +21,8 @@ _Warning: don't import `flow` from `"mobx"`, but from `"mobx-state-tree"` instea
 ```javascript
 import { types, flow } from "mobx-state-tree"
 
-someModel.actions(self => {
-    const fetchProjects = flow(function*() {
+someModel.actions((self) => {
+    const fetchProjects = flow(function* () {
```
Comment (coolsoftwaretyler) [inline docs/concepts/async-actions.md:25]: Looks like my format-on-save bumped this a little - should be fine.

File: docs/concepts/async-actions.md [hunk 2]
```diff
@@ -60,11 +60,12 @@ This means that each step in an asynchronous flow that needs to actually change
...
 
 ```javascript
-const Store = types.model({
+const Store = types
+    .model({
         githubProjects: types.array(types.frozen),
         state: types.enumeration("State", ["pending", "done", "error"])
     })
-    .actions(self => ({
+    .actions((self) => ({
```
Comment (coolsoftwaretyler) [inline docs/concepts/async-actions.md:68]: Looks like my format-on-save bumped this a little - should be fine.

File: docs/concepts/async-actions.md [hunk 3]
```diff
@@ -100,12 +100,14 @@ Generators might sound scary, but they are very suitable for expressing asynchro
...
+const Store = types
+    .model({
         githubProjects: types.array(types.frozen),
         state: types.enumeration("State", ["pending", "done", "error"])
     })
-    .actions(self => ({
-        fetchProjects: flow(function* fetchProjects() { // <- note the star, this a generator function!
+    .actions((self) => ({
+        fetchProjects: flow(function* fetchProjects() {
+            // <- note the star, this a generator function!
```
Comment (coolsoftwaretyler) [inline docs/concepts/async-actions.md:110]: Looks like my format-on-save bumped this a little - should be fine.

File: packages/mst-middlewares/README.md [hunk 4]
```diff
@@ -310,7 +310,7 @@ export const Store = types
 
         return {
             afterCreate() {
-                self.todos.push({ title: 'New Todo' })
+                self.todos.push({ title: "New Todo" })
```
Comment (coolsoftwaretyler) [inline packages/mst-middlewares/README.md:313]: Looks like my format-on-save bumped this a little - should be fine.

**Comment by coolsoftwaretyler** [comment]
Had a few format-on-save changes as well, but nothing major.

**Review by jamonholmgren — APPROVED** [review]
:tada:

**Pull request merged**